Fix Tune and Chrom Correction IOCs

anacso17 opened the pull request.

fernandohds564 commented on as-ap-opticscorr/as_ap_opticscorr/tune/main.py:
```diff
@@ -465,114 +480,125 @@ def _callback_estimate_deltatune(self, pvname, value, **kws):
-            self._timing_check_config[4] = (1 if value == 3 else 0)  # External
-        elif 'TunesDelay' in pvname:
-            self._timing_check_config[5] = (1 if value == 0 else 0)  # 0s
+            self._timing_check_config[3] = (value == 0.15)  # 150us
```
Ana, the `Duration` PV from timing is in `us` now. You should change the comparison here to `value == 150`

fernandohds564 commented on as-ap-opticscorr/as_ap_opticscorr/chrom/main.py:
```diff
@@ -427,78 +429,71 @@ def _callback_estimate_chrom(self, pvname, value, **kws):
+            self._timing_check_config[2] = (value == 1)  # 1 pulse
         elif 'Sexts:Duration' in pvname:
-            self._timing_check_config[3] = (1 if value == 0.15 else 0)  # 150us
+            self._timing_check_config[3] = (value == 0.15)  # 150us
```
The same here.

fernandohds564 commented on as-ap-opticscorr/as_ap_opticscorr/chrom/main.py:
```diff
@@ -427,78 +429,71 @@ def _callback_estimate_chrom(self, pvname, value, **kws):
             self._timing_check_config[0] = value  # Enbl
         elif 'Sexts:EVGParam' in pvname:
-            self._timing_check_config[1] = (1 if value == 1 else 0)  # Chroms
+            self._timing_check_config[1] = (value == 1)  # Chroms
```
Please, use information from the `siriuspy.csdevice.timesys` module to make this comparison. Do not use a hard coded value, because this index can change.

fernandohds564 commented on as-ap-opticscorr/as_ap_opticscorr/chrom/main.py:
```diff
@@ -427,78 +429,71 @@ def _callback_estimate_chrom(self, pvname, value, **kws):
+            self._timing_check_config[4] = (value == 3)  # External
         elif 'ChromsDelay' in pvname:
-            self._timing_check_config[5] = (1 if value == 0 else 0)  # 0s
+            self._timing_check_config[5] = (value == 0)  # 0s
```
The comment you have made in this line should be `# 0us`, to explicitly inform the PV unit.

anacso17 commented:
@fernandohds564 thank you! I implemented your suggestions, changed si-ap-chromcorr to use TI csdevice and to consider an event called 'ChromSI', in case we consider using syncmode in the future.
